### PR TITLE
8307955: Prefer to PTRACE_GETREGSET instead of PTRACE_GETREGS in method 'ps_proc.c::process_get_lwp_regs'

### PR DESCRIPTION
**Pull request by lgxbslgx**
Hi all,

This patch revises the code of `ps_proc.c::process_get_lwp_regs`
to use `PTRACE_GETREGSET` first instead of `PTRACE_GETREGS`.
The `PTRACE_GETREGS` is not present on all architectures as the man page states [1].
And if we use `PTRACE_GETREGS` first, several tests will fail at the special envs,
such as my local riscv64-linux env. Please see [the issue](https://bugs.openjdk.org/browse/JDK-8307955) for more information.

And I remove the unnecessary comments and macro,
because the `sparc` arch related code had been removed.

Thanks for the review.

Best Regards,
-- Guoxiong

[1] https://man7.org/linux/man-pages/man2/ptrace.2.html

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8307955](https://bugs.openjdk.org/browse/JDK-8307955): Prefer to PTRACE_GETREGSET instead of PTRACE_GETREGS in method 'ps_proc.c::process_get_lwp_regs'

### Reviewers
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)
 * [Kevin Walls](https://openjdk.org/census#kevinw) (@kevinjwalls - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13939/head:pull/13939` \
`$ git checkout pull/13939`

Update a local copy of the PR: \
`$ git checkout pull/13939` \
`$ git pull https://git.openjdk.org/jdk.git pull/13939/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13939`

View PR using the GUI difftool: \
`$ git pr show -t 13939`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13939.diff">https://git.openjdk.org/jdk/pull/13939.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13939#issuecomment-1544500826)

### CODE REVIEW

**Review by plummercj — APPROVED** [review]
Changes look good. I tested with our internal test system on all supported platforms and did not see any issues.

**Comment by lgxbslgx** [comment]
@plummercj Thanks for your test and review.

Waiting for another review.

**Review by kevinjwalls — APPROVED** [review]
Looks good to me.

**Comment by lgxbslgx** [comment]
@plummercj @kevinjwalls Thanks for the reviews. Integrating.
/integrate

**Pull request closed without merging**
